language/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -652,7 +652,65 @@ impl CompiledPackage {
         Ok((compiled_package, CompilationCachingStatus::Recompiled))
     }
 
+    // We take the (restrictive) view that all filesystems are case insensitive to maximize
+    // portability of packages.
+    fn check_filepaths_ok(&self) -> Result<()> {
+        // A mapping of (lowercase_name => (real_name, [info_for_each_occurence])
+        let mut insensitive_mapping = BTreeMap::new();
+        for compiled_unit in &self.compiled_units {
+            let is_module = matches!(&compiled_unit.unit, CompiledUnit::Module(_));
+            let name = match &compiled_unit.unit {
+                CompiledUnit::Script(named) => named.name.as_str(),
+                CompiledUnit::Module(named) => named.name.as_str(),
+            };
+            let entry = insensitive_mapping
+                .entry(name.to_lowercase())
+                .or_insert_with(BTreeMap::new);
+            let insert = entry.entry(name).or_insert_with(Vec::new);
+            insert.push((
+                is_module,
+                compiled_unit.source_path.to_string_lossy().to_string(),
+            ));
+        }
+        let errs = insensitive_mapping
+            .into_iter()
+            .filter_map(|(insensitive_name, preimages)| {
```

Comment:
I am confused why it is called preimages?

Reply:
Ah I was thinking of these as being the preimage of `insensitive_name` (i.e., all the names that mapped to it under `to_lowercase`). I'll update with a better name.